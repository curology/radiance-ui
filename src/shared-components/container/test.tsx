import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { Container } from './index';

describe('Container UI snapshots', () => {
  it('renders basic container', () => {
    const { container } = render(<Container>Container Content</Container>);

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it('renders container helper components', () => {
    const { container } = render(
      <Container>
        <Container.Section>Section 1</Container.Section>
        <Container.Divider />

        <Container.Image src="https://images-na.ssl-images-amazon.com/images/I/61E0pFPOl7L._SL1500_.jpg" />
        <Container.Section>
          <div>Section 2</div>
        </Container.Section>
      </Container>,
    );

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it('renders clickable container', () => {
    const { container } = render(
      <Container type="clickable">Container Content</Container>,
    );

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it('renders message type container', () => {
    const { container } = render(
      <Container type="message">Container Content</Container>,
    );

    expect(container.firstElementChild).toMatchSnapshot();
  });
});
