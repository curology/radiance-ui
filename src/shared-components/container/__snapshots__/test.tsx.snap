// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container UI snapshots renders basic container 1`] = `
.emotion-0 {
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
}

<div
  class="emotion-0 emotion-1"
>
  Container Content
</div>
`;

exports[`Container UI snapshots renders clickable container 1`] = `
.emotion-0 {
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.05);
  cursor: pointer;
  -webkit-transition: box-shadow 200ms;
  transition: box-shadow 200ms;
}

.emotion-0:hover {
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.10);
}

<div
  class="emotion-0 emotion-1"
  type="clickable"
>
  Container Content
</div>
`;

exports[`Container UI snapshots renders container helper components 1`] = `
.emotion-0 {
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
}

.emotion-2 {
  padding: 1.5rem;
}

@media (min-width: 992px) {
  .emotion-2 {
    padding-left: 2.5rem;
    padding-right: 2.5rem;
  }
}

.emotion-4 {
  margin: 0 1.5rem;
  border-bottom: 1px solid #EDEDF0;
}

@media (min-width: 992px) {
  .emotion-4 {
    margin: 0 2.5rem;
  }
}

.emotion-6 {
  width: 100%;
  overflow: hidden;
  object-fit: cover;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    Section 1
  </div>
  <div
    class="emotion-4 emotion-5"
  />
  <img
    class="emotion-6 emotion-7"
    src="https://images-na.ssl-images-amazon.com/images/I/61E0pFPOl7L._SL1500_.jpg"
  />
  <div
    class="emotion-2 emotion-3"
  >
    <div>
      Section 2
    </div>
  </div>
</div>
`;

exports[`Container UI snapshots renders message type container 1`] = `
.emotion-0 {
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
  border-radius: 8px;
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.05);
}

<div
  class="emotion-0 emotion-1"
  type="message"
>
  Container Content
</div>
`;
